net/validators_test.go

```diff
@@ -0,0 +1,128 @@
+package net_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	logging "github.com/ipfs/go-log"
+	"github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p-pubsub/pb"
+	mocknet "github.com/libp2p/go-libp2p/p2p/net/mock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/go-filecoin/address"
+	"github.com/filecoin-project/go-filecoin/chain"
+	"github.com/filecoin-project/go-filecoin/clock"
+	"github.com/filecoin-project/go-filecoin/consensus"
+	"github.com/filecoin-project/go-filecoin/net"
+	th "github.com/filecoin-project/go-filecoin/testhelpers"
+	tf "github.com/filecoin-project/go-filecoin/testhelpers/testflags"
+	"github.com/filecoin-project/go-filecoin/types"
+)
+
+func init() {
+	logging.SetDebugLogging()
+}
+
+func TestBlockTopicValidator(t *testing.T) {
+	tf.UnitTest(t)
+
+	ctx := context.Background()
+	mbv := th.NewMockBlockValidator()
+	tv := net.NewBlockTopicValidator(mbv, nil)
+	builder := chain.NewBuilder(t, address.Undef)
+	pid1 := th.RequireIntPeerID(t, 1)
+
+	goodBlk := builder.BuildOnBlock(nil, func(b *chain.BlockBuilder) {})
+	badBlk := builder.BuildOnBlock(nil, func(b *chain.BlockBuilder) {
+		b.IncHeight(1)
+	})
+
+	mbv.StubSyntaxValidationForBlock(badBlk, fmt.Errorf("invalid block"))
+
+	validator := tv.Validator()
+
+	assert.Equal(t, net.BlockTopic, tv.Topic())
+	assert.True(t, validator(ctx, pid1, blkToPubSub(goodBlk)))
+	assert.False(t, validator(ctx, pid1, blkToPubSub(badBlk)))
+	assert.False(t, validator(ctx, pid1, nonBlkPubSubMsg()))
+}
+
+func TestBlockPubSubValidation(t *testing.T) {
+	tf.IntegrationTest(t)
```

Comment:
TODO before merging file an issues as this test has some issues since pubsub messages are not guaranteed to arrive in order.